### PR TITLE
New files for Tasks API - beta

### PR DESCRIPTION
**Pull request by avijityadav**

### CODE REVIEW

File: api-reference/beta/api/basetask-update.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+}
+```
+
+
+### Response
+**Note:** The response object shown here might be shortened for readability.
+<!-- {
+  "blockType": "response",
+  "truncated": true
+}
```
Comment (isvargasmsft) [inline api-reference/beta/api/basetask-update.md:114]: Please add the odata type of the response. Please apply to all files and examples in this PR.

Comment (avijityadav) [reply api-reference/beta/api/basetask-update.md:114]: Sure Isaac, Looking into it.

File: api-reference/beta/api/basetask-update.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  }
+}
+```
+
+
+### Response
+**Note:** The response object shown here might be shortened for readability.
+<!-- {
+  "blockType": "response",
+  "truncated": true
```
Comment (isvargasmsft) [inline api-reference/beta/api/basetask-update.md:114]: odata type still missing here.

Comment (avijityadav) [reply api-reference/beta/api/basetask-update.md]: @isvargasmsft , Fixed it now, missed adding that file.

File: api-reference/beta/resources/basetask.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+doc_type: resourcePageType
+---
+
+# baseTask resource type
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents a task, such as a piece of work or personal item, that can be tracked and completed.
```
Comment (isvargasmsft) [inline api-reference/beta/resources/basetask.md:16]: Lines 3 and 16: Search engines use this description to display search results. Please make both lines (3 and 16) consistent (use the same description in both lines) and keep in mind this is used by search engines, so phrase it in the way you would (Bing) search this topic.

Comment (avijityadav) [reply api-reference/beta/resources/basetask.md:16]: Sure @isvargasmsft , I have kept the description same in both places. Yes, we have already used this definition to define tasks earlier. So, it would be fine if search engines use this description.

File: api-reference/beta/resources/basetasklist.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+|[List tasks](../api/basetasklist-list-tasks.md)|[baseTask](../resources/basetask.md) collection|Get the baseTask resources from the tasks navigation property.|
+|[Create baseTask](../api/basetasklist-post-tasks.md)|[baseTask](../resources/basetask.md)|Create a new baseTask object.|
+
+The following methods apply only to **taskList** type of derived task list.
+
+|Method|Return type|Description|
+|:---|:---|:---|
+|[Create baseTaskList](../api/tasks-post-lists.md)|[baseTaskList](../resources/basetasklist.md)|Create a new [baseTaskList](../resources/basetasklist.md) object.|
+|[Update baseTaskList](../api/basetasklist-update.md)|[baseTaskList](../resources/basetasklist.md)|Update the properties of a [baseTaskList](../resources/basetasklist.md) object.|
+|[Delete baseTaskList](../api/basetasklist-delete.md)|None|Deletes a [baseTaskList](../resources/basetasklist.md) object.|
```
Comment (isvargasmsft) [inline api-reference/beta/resources/basetasklist.md:40]: I'd move these methods to the specific topic of **taskList** resource type, leaving in the **baseTaskList** topic only the general qualities that apply to all children types.

File: api-reference/beta/resources/tasklist.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+doc_type: resourcePageType
+---
+
+# taskList resource type
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+A user created list in Microsoft To Do that contains one or more [Task](./task.md) resources.
```
Comment (isvargasmsft) [inline api-reference/beta/resources/tasklist.md:16]: Lines 3 and 16: This description is not easy to understand by a global audience. "A user created list" is the conflicting part. 

I'd suggest: Represents a list created by a user in Microsoft To Do that contains one or more...

File: api-reference/beta/resources/linkedresource_v2.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+# linkedResource_v2 resource type
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents an item in a partner application related to a [baseTask](./basetask.md). An example is an email from where the task was created. A **linkedResource** object stores information about that source application, and lets you link back to the related item. You can see the **linkedResource** in the task details view, as shown.
+
+![Linked resource in task details pane](/graph/images/todo-linkedresource-taskdetail.png)
```
Comment (isvargasmsft) [inline api-reference/beta/resources/linkedresource_v2.md:18]: Please add alt text to all the images.

File: api-reference/beta/api/basetask-list-checklistitems.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+ms.prod: "outlook"
+doc_type: apiPageType
+---
+
+# List checklistItems
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Get the checklistItem resources from the checklistItems navigation property.
```
Comment (isvargasmsft) [inline api-reference/beta/api/basetask-list-checklistitems.md:15]: Please add the reference link to **checkListItem** resource type in the first mention and bold subsequent mentions.

File: api-reference/beta/api/basetask-list-linkedresources.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+ms.prod: "outlook"
+doc_type: apiPageType
+---
+
+# List linkedResources
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Get the linkedResource_v2 resources from the linkedResources navigation property.
```
Comment (isvargasmsft) [inline api-reference/beta/api/basetask-list-linkedresources.md:15]: Please add the reference link to the first mention of linkedResource_v2 resource type and bold subsequent mentions.

File: api-reference/beta/api/basetask-list-linkedresources.md [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+  "blockType": "ignored"
+}
+-->
+``` http
+GET /me/tasks/lists/{baseTaskListId}/tasks/{baseTaskId}/linkedResources
+GET /users/{id | userPrincipalName}/tasks/lists/{baseTaskListId}/tasks/{baseTaskId}/linkedResources
+```
+
+## Optional query parameters
+This method supports some of the OData query parameters to help customize the response. For general information, see [OData query parameters](/graph/query-parameters).
```
Comment (isvargasmsft) [inline api-reference/beta/api/basetask-list-linkedresources.md:38]: It is always appreciated if you can mention which query parameters are supported.

File: api-reference/beta/api/basetask-move.md [hunk 10]
```diff
@@ -0,0 +1,112 @@
+---
+title: "Move baseTask"
+description: "Move a baseTask object from one baseTaskList to another."
+author: "devindrajit"
+ms.localizationpriority: medium
+ms.prod: "outlook"
+doc_type: apiPageType
+---
+
+# Move baseTask
```
Comment (isvargasmsft) [inline api-reference/beta/api/basetask-move.md:10]: Lines 2 and 10, please revise title to: **baseTask: move**

File: api-reference/beta/api/basetask-post-checklistitems.md [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+ms.prod: "outlook"
+doc_type: apiPageType
+---
+
+# Create checklistItem
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Create a new checklistItem object.
```
Comment (isvargasmsft) [inline api-reference/beta/api/basetask-post-checklistitems.md:15]: Please add the link to checkListItem.

File: api-reference/beta/api/basetasklist-delete.md [hunk 12]
```diff
@@ -0,0 +1,72 @@
+---
+title: "Delete baseTaskList"
```
Comment (isvargasmsft) [inline api-reference/beta/api/basetasklist-delete.md:2]: I understand you cannot delete a **wellKnownTaskList** resource, so I believe this method belongs to **taskLisk** instead of **baseTaskList**. 

Please clarify the functionality of this method.

File: api-reference/beta/api/basetasklist-list-tasks.md [hunk 13]
```diff
@@ -0,0 +1,100 @@
+---
+title: "List basetasks"
```
Comment (isvargasmsft) [inline api-reference/beta/api/basetasklist-list-tasks.md:2]: Please make this title consistent with the title in line 10.

File: api-reference/beta/api/basetasklist-update.md [hunk 14]
```diff
@@ -0,0 +1,95 @@
+---
+title: "Update baseTaskList"
```
Comment (isvargasmsft) [inline api-reference/beta/api/basetasklist-update.md:2]: Similar observation above: Please clarify if this method belongs to baseTaskList or taskList. I understand you cannot update a wellKnownTaskList.

File: api-reference/beta/api/tasks-list-lists.md [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+ms.prod: "outlook"
+doc_type: apiPageType
+---
+
+# List lists
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Get a list of the [baseTaskList](../resources/basetasklist.md) objects and their properties.
```
Comment (isvargasmsft) [inline api-reference/beta/api/tasks-list-lists.md:15]: In this description, I'd like to specify that we are listing the baseTaskLists objects of a user.

File: api-reference/beta/resources/basetasklist.md (hunk 4 above)
Comment (avijityadav) [reply api-reference/beta/resources/basetasklist.md:40]: Got it, make sense. I will do it.

File: api-reference/beta/api/basetasklist-delete.md (hunk 12 above)
Comment (avijityadav) [reply api-reference/beta/api/basetasklist-delete.md:2]: It makes sense.
So, we will be making the above change for Create, Update and Delete methods. Changed the title and description from **baseTaskList** to **taskList** in related files.

File: api-reference/beta/api/basetasklist-list-tasks.md (hunk 13 above)
Comment (avijityadav) [reply api-reference/beta/api/basetasklist-list-tasks.md:2]: Changed the title to List baseTasks in both places.

File: api-reference/beta/api/tasks-list-lists.md (hunk 15 above)
Comment (avijityadav) [reply api-reference/beta/api/tasks-list-lists.md:15]: Done.

File: api-reference/beta/resources/linkedresource_v2.md (hunk 6 above)
Comment (avijityadav) [reply api-reference/beta/resources/linkedresource_v2.md]: done

File: api-reference/beta/api/basetask-list-checklistitems.md [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+ms.prod: "outlook"
+doc_type: apiPageType
+---
+
+# List checklistItems
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Get the [checklistItem](../resources/checklistitem.md) resources from the checklistItems navigation property.
```
Comment (isvargasmsft) [inline api-reference/beta/api/basetask-list-checklistitems.md:15]: Lines 3 and 15, I'd suggest a description clarifying the relationship instead of just mentioning it. For example:
Get the checklistItem resources associated to a baseTask.

Comment (isvargasmsft) [reply api-reference/beta/api/basetask-list-checklistitems.md:15]: Please include reference link to both resources in their first mention.

File: api-reference/beta/api/basetask-list-linkedresources.md [hunk 17]
```diff
@@ -0,0 +1,89 @@
...
+ms.prod: "outlook"
+doc_type: apiPageType
+---
+
+# List linkedResources
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Get the [linkedResource_v2](../resources/linkedresource_v2.md) resources from the linkedResources navigation property.
```
Comment (isvargasmsft) [inline api-reference/beta/api/basetask-list-linkedresources.md:15]: Similar observation above regarding the description for List methods: Please rephrase to include a reference to both resources. For example:
Get the linkedResource_v2 resources associated to a baseTask.

File: api-reference/beta/api/basetasklist-post-tasks.md [hunk 18]
```diff
@@ -0,0 +1,138 @@
...
+ms.prod: "outlook"
+doc_type: apiPageType
+---
+
+# Create baseTask
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Create a new baseTask object in a specified [baseTaskList](../resources/basetasklist.md).
```
Comment (isvargasmsft) [inline api-reference/beta/api/basetasklist-post-tasks.md:15]: Lines 3 and 15: please rephrase to:

Create a new baseTask object <ins> in a specific </ins> baseTaskList. 

Please add the reference link to both resource types.

File: api-reference/beta/api/basetask-list-checklistitems.md (hunk 16 above)
Comment (avijityadav) [reply api-reference/beta/api/basetask-list-checklistitems.md:15]: Making the change.

File: api-reference/beta/api/basetask-list-linkedresources.md (hunk 17 above)
Comment (avijityadav) [reply api-reference/beta/api/basetask-list-linkedresources.md:15]: made the change.

File: api-reference/beta/api/tasks-list-lists.md [hunk 19]
```diff
@@ -0,0 +1,89 @@
+---
+title: "List lists"
+description: "Get a list of the baseTaskList objects and their properties."
```
Comment (isvargasmsft) [inline api-reference/beta/api/tasks-list-lists.md:3]: Lines 3 and 15: please remember both descriptions should match exactly.

Comment (avijityadav) [reply api-reference/beta/api/tasks-list-lists.md:3]: Oh, Sure, Will do that.

File: api-reference/beta/resources/tasks-overview.md [hunk 20]
```diff
@@ -0,0 +1,65 @@
...
+* Create tasks from your app’s workflow, for example, from email or notifications, and save them in To Do. Use the [linkedResource](linkedresource.md) entity to store the link back to your app.
+* Sync your app’s existing tasks with To Do and create a single task view for better prioritization and manageability.
+* Manage To Do tasks in a custom business application.
+* Create [checklistItems](checklistitem.md) on a task to break down complex tasks in smaller steps.
+
+Currently, the API supports only permissions delegated by the signed-in user.
+
+Before starting with the To Do Tasks API, take a look at the resources and how they relate to one another.
+
+![Screenshot highlighting To Do tasks API entities. On the left, we show list of taskList resources and one taskList is highlighted. In the center, we show a list of tasks of which one is highlighted. On the right, we show task details pain in which checklist Item and linked resource are highlighted.](/graph/images/tasks-api-entities.png)
```
Comment (isvargasmsft) [inline api-reference/beta/resources/tasks-overview.md:25]: What do you mean by task details pain? Can you rephrase to make it clearer?

File: api-reference/beta/resources/wellknowntasklist.md [hunk 21]
```diff
@@ -0,0 +1,69 @@
...
+doc_type: resourcePageType
+---
+
+# wellKnownTaskList resource type
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+A built-in task list which cannot be renamed or deleted. To Do has two built-in lists, **Flagged emails** and **Tasks** list.
```
Comment (isvargasmsft) [inline api-reference/beta/resources/wellknowntasklist.md:16]: Flagged emails: please don't capitalize unless mentioning a TMd name or brand. For example, Microsoft To Do. Please remove the capitalization of "flagged emails" and apply this to all files in this PR where there are capitalized words that don't correspond to a TMd name.

File: changelog/Microsoft.Tasks.json [hunk 22]
```diff
@@ -0,0 +1,256 @@
+﻿{
+  "changelog": [
+    {
+      "ChangeList": [
+        {
+          "Id": "b9609297-c6ba-42dd-a27f-bd9e1b800898",
```
Comment (isvargasmsft) [inline changelog/Microsoft.Tasks.json:6]: Were these changelog entries automatically or manually generated? If they were automatically generated we should open a bug to report some inconsistencies. 

Guidelines for changelog entries: https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/103153/Changelog?anchor=changelist-record-properties

File: changelog/Microsoft.Tasks.json [hunk 23]
```diff
@@ -0,0 +1,256 @@
...
+          "Description": "Added the [recurrencePatternType](https://docs.microsoft.com/en-us/?view=graph-rest-beta) resource",
+          "Target": "recurrencePatternType"
+        },
+        {
+          "Id": "b9609297-c6ba-42dd-a27f-bd9e1b800898",
+          "ApiChange": "EnumType",
+          "ChangedApiName": "recurrenceRangeType",
+          "ChangeType": "Addition",
+          "Description": "Added the [recurrenceRangeType](https://docs.microsoft.com/en-us/?view=graph-rest-beta) resource",
+          "Target": "recurrenceRangeType"
```
Comment (isvargasmsft) [inline changelog/Microsoft.Tasks.json:43]: Lines 7 to 43: I dont see these enums in the enums.md file. Could you please clarify?

File: changelog/Microsoft.Tasks.json [hunk 24]
```diff
@@ -0,0 +1,256 @@
...
+          "Description": "Added the [taskStatus_v2](https://docs.microsoft.com/en-us/?view=graph-rest-beta) resource",
+          "Target": "taskStatus_v2"
+        },
+        {
+          "Id": "b9609297-c6ba-42dd-a27f-bd9e1b800898",
+          "ApiChange": "EnumType",
+          "ChangedApiName": "weekIndex",
+          "ChangeType": "Addition",
+          "Description": "Added the [weekIndex](https://docs.microsoft.com/en-us/?view=graph-rest-beta) resource",
+          "Target": "weekIndex"
```
Comment (isvargasmsft) [inline changelog/Microsoft.Tasks.json:59]: I dont see this enum in the enum.md file. Could you please clarify?

File: changelog/Microsoft.Tasks.json [hunk 25]
```diff
@@ -0,0 +1,256 @@
...
+          "Id": "b9609297-c6ba-42dd-a27f-bd9e1b800898",
+          "ApiChange": "EnumType",
+          "ChangedApiName": "recurrenceRangeType",
+          "ChangeType": "Addition",
+          "Description": "Added the [recurrenceRangeType](https://docs.microsoft.com/en-us/?view=graph-rest-beta) resource",
+          "Target": "recurrenceRangeType"
+        },
+        {
+          "Id": "b9609297-c6ba-42dd-a27f-bd9e1b800898",
+          "ApiChange": "EnumType",
```
Comment (isvargasmsft) [inline changelog/Microsoft.Tasks.json:47]: ApiChange should be "Enum type", with a space, as seeing here: https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/103153/Changelog?anchor=changelist-record-examples

File: changelog/Microsoft.Tasks.json [hunk 26]
```diff
@@ -0,0 +1,256 @@
...
+          "ChangedApiName": "taskStatus_v2",
+          "ChangeType": "Addition",
+          "Description": "Added the [taskStatus_v2](https://docs.microsoft.com/en-us/?view=graph-rest-beta) resource",
```
Comment (isvargasmsft) [inline changelog/Microsoft.Tasks.json:50]: Please rephrase to: Added the taskStatus enumeration. Delete the link because enums dont have resource topic and just bold the mention of the enum.

Comment (isvargasmsft) [reply changelog/Microsoft.Tasks.json:50]: Please apply this comment to all enums in this Changelog entry.

File: changelog/Microsoft.Tasks.json (hunk 22 above)
Comment (avijityadav) [reply changelog/Microsoft.Tasks.json:6]: These were automatically generated using the method mentioned in above Link.

File: changelog/Microsoft.Tasks.json (hunk 25 above)
Comment (avijityadav) [reply changelog/Microsoft.Tasks.json]: Done.

File: changelog/Microsoft.Tasks.json (hunk 24 above)
Comment (avijityadav) [reply changelog/Microsoft.Tasks.json]: These are existing enums in enums.md and not added as part of PR. Since we are not adding new enums, I have removed these entries from changelog.

File: changelog/Microsoft.Tasks.json (hunk 23 above)
Comment (avijityadav) [reply changelog/Microsoft.Tasks.json]: These are existing enums in enums.md. Since we are not adding new enums, I have removed these entries from changelog.

File: api-reference/beta/resources/wellknowntasklist.md (hunk 21 above)
Comment (avijityadav) [reply api-reference/beta/resources/wellknowntasklist.md]: Done

File: api-reference/beta/resources/tasks-overview.md (hunk 20 above)
Comment (avijityadav) [reply api-reference/beta/resources/tasks-overview.md]: typo here "pane". I have rephrased it to make it easier to understand.

**Review by isvargasmsft — APPROVED** [review]
